tests/ui/vec_init_then_push.rs

```diff
@@ -44,3 +44,45 @@ pub fn no_lint() -> Vec<i32> {
         }
     }
 }
+
+fn _from_iter(items: impl Iterator<Item = u32>) -> Vec<u32> {
+    let mut v = Vec::new();
+    v.push(0);
+    v.push(1);
+    v.extend(items);
+    v
+}
```

Comment:
Could you add a second test like this, that reaches the threshold and triggers the lint? From my understanding, it should be as simple as:

```rs
fn _from_iter_trigger(items: impl Iterator<Item = u32>) -> Vec<u32> {
    let mut v = Vec::new();
    v.push(0);
    v.push(1);
    v.push(2);
    v.push(3);
    v.extend(items);
    v
}
```

Also, a test with the capacity would be good, like:

```rs
   let mut cap_err = Vec::with_capacity(4);
    cap_err.push(0);
    cap_err.push(1);
    cap_err.push(2);
```
That shouldn't lint either from my understanding

Reply:
The reason for avoiding the boundary case is to avoid having to change the test if we decide to nudge it a little. It doesn't really matter either way, just that the specific number isn't important.

Reply:
> The reason for avoiding the boundary case is to avoid having to change the test if we decide to nudge it a little.

Okay, then could you add one with some more space, it doesn't have to be at the boundary, but I don't see one for this case :upside_down_face:

Reply:
The last test covers it with a much larger number.

Reply:
The last test covers an if statement. The lint already took if statements into account, that's why I didn't consider this a test to this case. Since you want to add a note about the other comment, could you add a test with `extend` or some other method?